adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/AmqpContext.java

```diff
@@ -27,12 +28,20 @@
     private final Message message;
     private final ResourceIdentifier resource;
     private final Device authenticatedDevice;
+    private final Target receiverTarget;
 
-    AmqpContext(final ProtonDelivery delivery, final Message message, final ResourceIdentifier resource, final Device authenticatedDevice) {
+    AmqpContext(final ProtonDelivery delivery, final Message message, final Target receiverTarget, final Device authenticatedDevice) {
         this.delivery = delivery;
         this.message = message;
-        this.resource = resource;
         this.authenticatedDevice = authenticatedDevice;
+        this.receiverTarget = receiverTarget;
+        if (receiverTarget == null) {
+            // link possibly opened by an "authenticated" gateway component via anonymous relay
+            // TODO: validate the address in the message's _to property
+            this.resource = ResourceIdentifier.fromString(message.getProperties().getTo());
```

Comment:
what if *to* is empty?

Reply:
@sophokles73: this is really an amazing feedback, thanks! i will not respond to all the comments since I agree with everything said :-) will update the PR